lightningd/opening_control.c

```diff
@@ -804,13 +806,83 @@ static void openchannel_payload_remove_openingd(struct subd *openingd,
 	payload->openingd = NULL;
 }
 
-static void openchannel_hook_cb(struct openchannel_hook_payload *payload STEALS,
-			    const char *buffer,
-			    const jsmntok_t *toks)
+static bool
+openchannel_hook_deserialize(struct openchannel_hook_payload *payload,
+			     const char *buffer,
+			     const jsmntok_t *toks)
 {
 	struct subd *openingd = payload->openingd;
-	const u8 *our_upfront_shutdown_script;
-	const char *errmsg = NULL;
+
+	/* already rejected by prior plugin hook in the chain */
+	if (payload->errmsg != NULL)
+		return true;
+
+	if (!toks || !buffer)
+		return true;
+
+	const jsmntok_t *t = json_get_member(buffer, toks, "result");
+	if (!t)
+		fatal("Plugin returned an invalid response to the"
+		      " openchannel hook: %.*s",
+		      toks[0].end - toks[0].start,
+		      buffer + toks[0].start);
+
+	/* reject */
+	// TODO: hooks do not seem to be called serialized but parallel!
```

Comment:
Is this what's left on this PR?

Reply:
@niftynei It seems I have caused some valgrind stuff that needs to be addressed.

Reply:
The TODO was maybe caused by indeterminate code flow because of that. after that I will check if tests are finally stable.
I will close this draft and submit a fresh PR when done.